sotopia/database/utils.py

```diff
@@ -0,0 +1,181 @@
+    }
+
+
+def get_social_goals_from_episode(
+    epsidoes: list[EpisodeLog],
+) -> list[dict[str, str]]:
+    """Obtain social goals from episodes.
+
+    Args:
+        epsidoes (list[EpisodeLog]): List of episodes.
+
+    Returns:
+        list[dict[str, str]]: List of social goals with agent names as the index.
+    """
+    social_goals = []
+    for episode in epsidoes:
+        agents = [AgentProfile.get(agent) for agent in episode.agents]
+        agent_names = [
+            agent.first_name + " " + agent.last_name for agent in agents
+        ]
+        environment = EnvironmentProfile.get(episode.environment)
+        agent_goals = {
+            agent_names[0]: environment.agent_goals[0],
+            agent_names[1]: environment.agent_goals[1],
+        }
+        social_goals.append(agent_goals)
+    return social_goals
+
+
+def get_social_interactions_from_episode(
+    epsidoes: list[EpisodeLog],
+) -> list[str]:
+    """Obtain pure social interactions from episodes.
+    Args:
+        epsidoes (list[EpisodeLog]): List of episodes.
+    Returns:
+        list[str]: List of social interactions.
+    """
+    social_interactions = []
+    for episode in epsidoes:
+        assert isinstance(episode.tag, str)
+        list_of_social_interactions = episode.render_for_humans()[1]
+        if len(list_of_social_interactions) < 3:
+            continue
+        if "script" in episode.tag.split("_"):
+            overall_social_interaction = list_of_social_interactions[1:-3]
+        else:
+            overall_social_interaction = list_of_social_interactions[0:-3]
+            # only get the sentence after "Conversation Starts:\n\n"
+            starter_msg_list = overall_social_interaction[0].split(
+                "Conversation Starts:\n\n"
+            )
+            if len(starter_msg_list) < 3:
+                overall_social_interaction = list_of_social_interactions[1:-3]
+                # raise ValueError("The starter message is not in the expected format")
+            else:
+                overall_social_interaction[0] = starter_msg_list[-1]
+
+        social_interactions.append("\n\n".join(overall_social_interaction))
+    return social_interactions
+
+
+def episodes_to_csv(
+    episodes: list[EpisodeLog], filepath: str = "episodes.csv"
+) -> None:
+    """Save episodes to a csv file.
+
+    Args:
+        episodes (list[EpisodeLog]): List of episodes.
+        filepath (str, optional): The file path. Defaults to "episodes.csv".
+    """
+    data = {
+        "episode_id": [episode.pk for episode in episodes],
+        "scenario": [
+            get_scenario_from_episode(episode) for episode in episodes
+        ],
+        "codename": [
+            get_codename_from_episode(episode) for episode in episodes
+        ],
+        "agents_background": [
+            get_agents_background_from_episode(episode) for episode in episodes
+        ],
+        "social_goals": get_social_goals_from_episode(episodes),
```

Comment:
why these two are not following the same as previous ones? like key: [get...(episode) for episode in episodes]

Reply:
fixed!